public/app/features/trails/utils.ts

```diff
@@ -47,6 +50,21 @@ export function getMetricSceneFor(model: SceneObject): MetricScene {
   throw new Error('Unable to find trail');
 }
 
+export function getDatasourceForNewTrail(): string | undefined {
+  const mostRecentTrail = getTrailStore().recent[0];
+  const initialDS = mostRecentTrail?.resolve().state.$variables?.getByName(VAR_DATASOURCE)?.getValue();
+  let initialDSValue = undefined;
+  if (typeof initialDS === 'string') {
+    initialDSValue = initialDS;
+  } else {
+    const promDatasources = getDatasourceSrv().getList({ type: 'prometheus' });
+    if (config.defaultDatasource && promDatasources.find((mds) => mds.uid === config.defaultDatasource)) {
+      initialDSValue = config.defaultDatasource;
+    }
+  }
+  return initialDSValue;
+}
```

Comment:
```suggestion
export function getDatasourceForNewTrail(): string | undefined {
  const prevTrail = getTrailStore().recent[0];
  if (!prevTrail) {
    return config.defaultDatasource;
  }

  const prevDataSource = sceneGraph.interpolate(prevTrail.resolve(), VAR_DATASOURCE_EXPR);

  if (typeof prevDataSource === 'string' && prevDataSource.length > 0) {
    return prevDataSource;
  }

  const promDatasources = getDatasourceSrv().getList({ type: 'prometheus' });
  if (promDatasources.length > 0) {
    return promDatasources.find((mds) => mds.uid === config.defaultDatasource) ?? promDatasources[0].uid;
  }

  return undefined;
}
```

Maybe something like that makes sense? use last trail data source, if no data source there, take the default if it's prometheus, if default is not prometheus take first prometheus ds